sql/expression/comparison.go

```diff
@@ -50,21 +50,25 @@ func PreciseComparison(e sql.Expression) bool {
 				return true
 			}
 
+			// string type comparisons are exact
+			if types.IsText(left) && types.IsText(right) {
+				return true
+			}
+
 			if tupType, ok := right.(types.TupleType); ok {
-				if types.IsInteger(left) {
-					rightIsAllInt := true
-					for _, typ := range tupType {
-						if !types.IsInteger(typ) {
-							rightIsAllInt = false
-							break
+				for _, typCb := range []func(t sql.Type) bool{types.IsText, types.IsInteger} {
```

Comment:
Would pull out a tupleTypesMatch function, then this becomes

```go
if tupleTypesMatch(left, right, types.IsText) || tupleTypesMatch(left, right, types.IsInteger) {  return true }
```

Reply:
kk